modules/internal/ChapelLocale.chpl

```diff
@@ -48,8 +48,32 @@ module ChapelLocale {
     // The parent of the root locale is nil (by definition).
     const parent : locale;
 
-    // To be removed from the required interface once legacy code is adjusted.
-    // Modified in RootLocale.init().
+    pragma "no doc" var nPUsPhysAcc: int;    // HW cores, OS accessible
+    pragma "no doc" var nPUsPhysAll: int;    // HW cores, all
+    pragma "no doc" var nPUsLogAcc: int;     // HW threads, OS accessible
+    pragma "no doc" var nPUsLogAll: int;     // HW threads, all
+
+    /*
+      A *processing unit* or *PU* is an instance of the processor
+      architecture, basically the thing that executes instructions.
+      ``numPUs()`` tells how many of these are present on this locale.
+      If ``physical==true`` (the default) this is the number of physical
+      PUs, commonly known as *cores*.  Otherwise it is the number of
+      hardware threads (such as hyperthreads), which on a multithreaded
+      hardware architecture would typically be more numerous.  If
+      ``accessible==true`` (the default) then the count reflects only
+      those PUs the program has access to, due to limiting by the OS for
```

Comment:
I think both of those examples (maybe starting with the one that isn't specific to multi-locale / cluster computation) would be nice to add if it doesn't seem too pedantic/heavyweight.  I can't remember what formatting options are available to us, but if there were a way to set it off as being clearly explanatory information and not necessary to understanding the call, that might be nice.  Having it at the bottom of the comment (which is what looks like would happen naturally) would also be OK, though, in the "important stuff first" sense.